### PR TITLE
Dynamic script icons

### PR DESCRIPTION
**Pull request by invalidCards**
Puts down a framework for easily creating icons, based on [Remix Icons](https://github.com/Remix-Design/RemixIcon). The CSS isn't my proudest work ever, and I don't expect this to be pulled in its current form, but I am open to suggestions 😄

### CODE REVIEW

File: docs/framework.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ This path is automatically fed into `browser.runtime.getURL()`, so should start
 
 ## Metadata files
 
-Each module should be accompanied by a `.json` file of matching name; i.e. `example.js` should be accompanied by `example.json` in the same directory level. Right now, only two keys are supported; expect more to be added in future.
+Each module should be accompanied by a `.json` file of matching name; i.e. `example.js` should be accompanied by `example.json` in the same directory level. Right now, only five keys are supported; expect more to be added in future.
```
Comment (AprilSylph) [inline docs/framework.md:33]:
```suggestion
Each module should be accompanied by a `.json` file of matching name, i.e. `example.js` should be accompanied by `example.json` in the same directory level. Right now, four keys are supported.
```
(i know the `"preferences"` documentation is missing but the number may as well be accurate in the meantime)

File: src/browser/action.js [hunk 2]
```diff
@@ -49,17 +49,31 @@ async function renderScripts() {
...
+      iconInner.classList.add(icon, 'ri-fw');
+      iconInner.style.color = iconForeground;
+      icondiv.appendChild(iconInner);
+      fieldset.appendChild(icondiv);
+    }
+
     if (description) {
       const p = document.createElement('p');
       p.textContent = description;
+      if (icon) {
```
Comment (AprilSylph) [inline src/browser/action.js:74]:
```suggestion
      if (icon.class_name !== undefined) {
```

**Comment by AprilSylph** [comment]
is RemixIcons available as an npm package? if it is, it would be great to add it as a dependency and add it to `dev/copy-libs.sh`

**Comment by invalidCards** [comment]
[Yes it is](https://www.npmjs.com/package/remixicon)

**Comment by AprilSylph** [comment]
everything LGTM save the CSS, i'll try playing with it soon to see how it can be cleaned up!

**Pull request merged**